### PR TITLE
refactor: usage of icons

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Refactored the usage of icons.
Updated migration guide with steps and tips to replace icons with new icon component.
Removed svg sizing documentation.

Created the list of potential icons to create/redraw [here](https://docs.google.com/spreadsheets/d/1ZyBhHEmG7xsbV9XHrpFiIOsSMNkPy9XnlsbqKYb8fWo/edit?usp=sharing)

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: migration_guide/MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -479,3 +483,32 @@ Search for | Replace with
...
 `var(--lh-looser)`  | `var(--lh-500)`
+
+### Icons
+
+#### Replace icons with dialtone-vue icon component if possible
+
+Replace any custom usage of icon component with <dt-icon name="icon-name" size="icon-size" />
+Do a search in [Icon docs](https://dialpad.design/components/icon.html) to find an icon that matches your needs.
+For icon component documentation on props (naming and sizing), check 
+[Dialtone Vue - Icon component](https://vue.dialpad.design/?path=/story/components-icon--default)
```
Comment (braddialpad) [inline migration_guide/MIGRATION_GUIDE.md]: There is still a ton of code in Dialpad that uses backbone and can't use vue components, so we'll need to provide the css/html way of doing it as well. Just make sure we emphasize to use the vue components in all vue code.

File: migration_guide/MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -479,3 +483,32 @@ Search for | Replace with
 `var(--lh-normal)`  | `var(--lh-300)`
 `var(--lh-loose)`   | `var(--lh-400)`
 `var(--lh-looser)`  | `var(--lh-500)`
+
+### Icons
+
+#### Replace icons with dialtone-vue icon component if possible
+
+Replace any custom usage of icon component with <dt-icon name="icon-name" size="icon-size" />
+Do a search in [Icon docs](https://dialpad.design/components/icon.html) to find an icon that matches your needs.
```
Comment (braddialpad) [inline migration_guide/MIGRATION_GUIDE.md]: Lets say to check with your designer or ask in #dialtone if you are having trouble finding a replacement for an existing icon.

File: migration_guide/MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -479,3 +483,32 @@ Search for | Replace with
 `var(--lh-normal)`  | `var(--lh-300)`
 `var(--lh-loose)`   | `var(--lh-400)`
 `var(--lh-looser)`  | `var(--lh-500)`
+
+### Icons
+
+#### Replace icons with dialtone-vue icon component if possible
+
+Replace any custom usage of icon component with <dt-icon name="icon-name" size="icon-size" />
```
Comment (braddialpad) [inline migration_guide/MIGRATION_GUIDE.md]: wrap the code in a code tag

File: docs/.vuepress/theme/components/Navbar.vue [hunk 4]
```diff
@@ -51,9 +53,11 @@
           alt="Codepen Template"
           class="d-btn d-btn--muted d-tooltip--hover"
           href="https://codepen.io/pen/?template=BajJpwW"
+          target="_blank"
+          rel="noreferrer noopener"
         >
           <img
-            class="d-svg--size20"
+            class="d-icon d-icon--size-400"
```
Comment (francisrupert) [inline docs/.vuepress/theme/components/Navbar.vue:60]: Oh, we don't have a `<dt-icon>` for Github and Codepen. That's cool. Good catch on at least using `.d-icon` and `d-icon--size-400`.

File: migration_guide/MIGRATION_GUIDE.md [hunk 5]
```diff
@@ -166,6 +166,10 @@ New Color Ramp
 
 ---
 
+### Icon updates:
+
+- Deprecated .d-svg class
```
Comment (francisrupert) [inline migration_guide/MIGRATION_GUIDE.md]: Update to something like...

> Deprecated .d-svg class associated with the previous Material Design icon set. The previous icon set will remain available until it is sunset upon complete adoption.

File: migration_guide/MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -479,3 +483,32 @@ Search for | Replace with
 `var(--lh-normal)`  | `var(--lh-300)`
 `var(--lh-loose)`   | `var(--lh-400)`
 `var(--lh-looser)`  | `var(--lh-500)`
+
+### Icons
+
+#### Replace icons with dialtone-vue icon component if possible
```
Comment (francisrupert) [inline migration_guide/MIGRATION_GUIDE.md]: Make it clear that any non-Vue use of an icon can be updated with the raw HTML equivalent, i.e. 
`<svg class=" d-icon   d-icon--size-[SIZE]  d-icon--[ICON_NAME]....`"

File: migration_guide/MIGRATION_GUIDE.md (hunk 1 above)
Comment (juliodialpad) [reply migration_guide/MIGRATION_GUIDE.md]: I'll need to update dialtone-icons to include the svg files too, so they can import svg's from dialtone-icons repo.

File: migration_guide/MIGRATION_GUIDE.md [hunk 7]
```diff
@@ -479,3 +483,32 @@ Search for | Replace with
...
+`d-svg--size14` | `d-icon--size-100`
+`d-svg--size16` | `d-icon--size-200`
+`d-svg--size18` | `d-icon--size-300`
+`d-svg--size20` | `d-icon--size-400`
+`d-svg--size24` | `d-icon--size-500`
+`d-svg--size32` | `d-icon--size-600`
+`d-svg--size48` | `d-icon--size-700`
+`d-svg--size64` | `d-icon--size-800`
+
+Remove any usage of `d-svg` and `d-svg--*`.
```
Comment (francisrupert) [inline migration_guide/MIGRATION_GUIDE.md:527]: Need a note for any _custom CSS_ products may have done locally. That is, remove any contextual sizing of the icon in favor of only use Vue's `size` prop or (for raw HTML usage) the `d-icon--size-NNN`. For example, no longer can do something like `.foo .icon { width: xxx; height: yyy; }`.

Markdown section as ...
```
#### Product custom CSS
```

**Review by francisrupert — CHANGES_REQUESTED** [review]
Great start, particularly the migration guide.

Two key things I don't see fully complete yet, though I'm guessing they're still in progress.

* Some instances of `.d-svg` remain within components. I won't list them all, but a few examples: Chip, Notice, Modal.
* Updating CSS to no longer _contextually resize_, e.g. `.d-btn--sm .d-btn__icon .d-svg {width: [size]; height: [size]; }`

**Comment by francisrupert** [comment]
@juliodialpad, @braddialpad What do you think of pulling out the Migration Guide to its own PR and release that to Production just to get it out there?

**Comment by braddialpad** [comment]
> @juliodialpad, @braddialpad What do you think of pulling out the Migration Guide to its own PR and release that to Production just to get it out there?

If @juliodialpad thinks the additional changes will take a while then yes. Though it sounds like he still needs to make some changes to dialtone-icons before releasing the migration guide.

**Comment by juliodialpad** [comment]
> > @juliodialpad, @braddialpad What do you think of pulling out the Migration Guide to its own PR and release that to Production just to get it out there?
> 
> If @juliodialpad thinks the additional changes will take a while then yes. Though it sounds like he still needs to make some changes to dialtone-icons before releasing the migration guide.

Yes, I'll need to add svg icons to dialtone-icons, I'll create a PR to that ASAP.

**Review by francisrupert — COMMENTED** [review]
@juliodialpad I made comments directly on the **commit** by accident, but forgot they wouldn't appear as a review comment.

In case you weren't alerted to them, you can see them at https://github.com/dialpad/dialtone/commit/ac70a4aa78333014f750a342e26c41ce1de551dd

**Comment by juliodialpad** [comment]
> Great start, particularly the migration guide.
> 
> Two key things I don't see fully complete yet, though I'm guessing they're still in progress.
> 
> * Some instances of `.d-svg` remain within components. I won't list them all, but a few examples: Chip, Notice, Modal.
> * Updating CSS to no longer _contextually resize_, e.g. `.d-btn--sm .d-btn__icon .d-svg {width: [size]; height: [size]; }`

I didn't removed that CSS because it will break all the styling in the app, I prefer to maintain the styling until the icons are fully migrated, that way, the migration can be released iteratively and don't need to do it all at a time.

**Comment by francisrupert** [comment]
> I didn't removed that CSS because it will break all the styling in the app, I prefer to maintain the styling until the icons are fully migrated, that way, the migration can be released iteratively and don't need to do it all at a time.

Yeah, great point. Removing that CSS assumes 100% adoption of the new icons. Can you capture a backlog task so we can iteratively follow through on it.

**Comment by juliodialpad** [comment]
> > I didn't removed that CSS because it will break all the styling in the app, I prefer to maintain the styling until the icons are fully migrated, that way, the migration can be released iteratively and don't need to do it all at a time.
> 
> Yeah, great point. Removing that CSS assumes 100% adoption of the new icons. Can you capture a backlog task so we can iteratively follow through on it.

Sure, I'll do it

**Review by francisrupert — APPROVED** [review]

**Pull request merged**